packages/blocto/src/index.ts

```diff
@@ -0,0 +1,78 @@
+import type { WalletInit } from '@web3-onboard/common'
+
+function BloctoWallet(): WalletInit {
+  if (typeof window === 'undefined') return () => null
+  return () => {
+    return {
+      label: 'Blocto',
+      getIcon: async () => (await import('./icon.js')).default,
+      getInterface: async ({ chains }) => {
+        const { default: BloctoSDK } = await import('@blocto/sdk')
+
+        const { createEIP1193Provider } = await import('@web3-onboard/common')
+
+        const [defaultChain] = chains
+
+        const instance = new BloctoSDK({
+          ethereum: {
+            chainId: defaultChain.id,
+            rpc: defaultChain.rpcUrl
+          }
+        })
+        const bloctoProvider: any = instance.ethereum
+
+        const provider = createEIP1193Provider(bloctoProvider, {
+          eth_selectAccounts: null,
+          wallet_switchEthereumChain: async ({ params, baseRequest }) => {
+            const chain = chains.find(({ id }) => id === params[0].chainId)
+            if (!chain) throw new Error('chain must be set before switching')
+            const providerRpcurl = bloctoProvider.switchableNetwork[chain.id]?.rpc_url
```

Comment:
We can't unfortunately support conditional chaining of props for backward compatibility for older projects